src/server/object_services/md_store.js

```diff
@@ -447,6 +453,18 @@ class MDStore {
         let delete_marker;
         if (filter_delete_markers === true) delete_marker = null;
         else if (filter_delete_markers === false) delete_marker = true;
+        let size_filter;
```

Comment:
we need to test the performance of these queries and if new indexes are necessary

Reply:
See [here](https://github.com/noobaa/noobaa-core/blob/cc184f7fbff7690bf8f31062b60d4d370b62df0f/src/server/object_services/md_store.js#L409)